External/vulkan/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# -------------------------------------------------------------------------
+# Vulkan
+
+add_library(ExtVulkan INTERFACE)
+
+if (NOT DEFINED ENV{VULKAN_SDK})
+    message(FATAL_ERROR "A 'VULKAN_SDK' environment variable is required")
+endif()
+
+if (CMP_HOST_APPLE)
+
+    set(MOLTEN_VK_PATHS
+
+        ${VULKAN_SDK}/macOS/framework
+        #${VULKAN_SDK}/macOS/static
+        #${VULKAN_SDK}/macOS/dynamic
+    )
+
+    find_library(
+        MOLTEN_VK_FRAMEWORK
+
+        NAMES MoltenVK
+        PATHS ${MOLTEN_VK_PATHS}
+        NO_DEFAULT_PATH
+    )
+
+    find_library(IO_KIT_FRAMEWORK		IOKit)
+    find_library(IO_SURFACE_FRAMEWORK	IOSurface)
+    find_library(METAL_FRAMEWORK		Metal)
+    find_library(QUARTZ_CORE_FRAMEWORK	QuartzCore)
+
+    set(
+        Vulkan_LIBRARIES
+
+        ${IO_KIT_FRAMEWORK}
+        ${IO_SURFACE_FRAMEWORK}
+        ${METAL_FRAMEWORK}
+        ${MOLTEN_VK_FRAMEWORK}
+        ${QUARTZ_CORE_FRAMEWORK}
+    )
+
+    find_path(MVK_INCLUDE_DIR
+        NAMES MoltenVK/mvk_vulkan.h
+        PATHS ${VULKAN_SDK}/include
+    )
+
+    set(
+        Vulkan_INCLUDE_DIRS
+
+        ${MVK_INCLUDE_DIR}
+        ${MOLTEN_VK_FRAMEWORK}
+    )
+
+elseif(CMP_HOST_LINUX)
+
+    # More details can be found on this issue: https://vulkan.lunarg.com/issue/view/5ed6eca8e0d2a37f6e19cefc
```

Comment:
There's likely a much better way to get GLSLANG but I wanted to use the same code that comes bundled with Vulkan. There's however, some bad build artifacts in the Vulkan SDK that prevent this. This is a hacky work-around for that.